pkg/inst/worker/daemon.R

```diff
@@ -0,0 +1,57 @@
+# Worker daemon
+
+rLibDir <- Sys.getenv("SPARKR_RLIBDIR")
+script <- paste(rLibDir, "SparkR/worker/worker.R", sep = "/")
+
+# preload SparkR package, speedup worker
+.libPaths(c(rLibDir, .libPaths()))
+suppressPackageStartupMessages(library(SparkR))
+
+port <- as.integer(Sys.getenv("SPARKR_WORKER_PORT"))
+inputCon <- socketConnection(port = port, open = "rb", blocking = TRUE, timeout = 3600)
+
+# Read from connection, retrying to read exactly 'size' bytes
+readBinFull <- function(con, size) {
+  # readBin() could be interruptted by signal, we have no way to tell
+  # if it's interruptted or the socket is closed, so we retry at most
+  # 20 times, to avoid the deadloop if the socket is closed
+  c <- 1
```

Comment:
I think this approach is fine, but I don't think the signal interruption concern is valid.  For example you can see in the R source code [1] that when the return code is `< 0` R raises an error (we could probably handle that with a tryCatch ?)

[1] https://github.com/wch/r-source/blob/354e3abe9574f4f830cb983bd1f6720e6e05e852/src/main/connections.c#L3985